### DISCUSSION TITLE
训练V3识别模型，一开启训练就爆显存

### DISCUSSION THREAD

**Discussion by Alanhzl** [PaddlePaddle/PaddleOCR#12766] [Q&A] [answered]
使用的是4090D，显存24G，在开启训练后显存占用从500M+升到24100M+，然后训练信息打印一次左右就报错：
`Out of memory error on GPU 0. Cannot allocate 129.394531MB memory on GPU 0, 23.586792GB memory has been allocated and available memory is only 49.250000MB.`

使用的配置文件是ch_PP-OCRv3_rec_distillation.yml
修改点：
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/39329022/eaf970e1-8b67-4e57-a810-ef121035da8a)
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/39329022/cc3fc4ed-c9c7-43c2-ab44-7cc9d7a5d9fc)
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/39329022/4f2e636b-104b-4c33-8203-380f864eafcf)

另外想问一下：
![image](https://github.com/PaddlePaddle/PaddleOCR/assets/39329022/8c1e5e19-de76-486f-b2e7-f9107897a54d)
这里是不是表示开启了五组学生教师模型？是不是这里导致爆显存？如果是的话应该怎么改？

**Comment by SWHL** [comment, marked as answer]
batch_size是不是太大了

https://github.com/PaddlePaddle/PaddleOCR/blob/862b4d46edc55b8bf3d7806e7c316e11a1d5f478/configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml#L182

**Reply by Alanhzl** [reply]
> batch_size是不是太大了
> 
> https://github.com/PaddlePaddle/PaddleOCR/blob/862b4d46edc55b8bf3d7806e7c316e11a1d5f478/configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml#L182

和这个配置文件一样，batch_size_per_card: 128，num_workers: 4

**Reply by SWHL** [reply]
这个配置训练所用机器显存可能比你的大，你尝试改成1试试

**Reply by Alanhzl** [reply]
> > batch_size是不是太大了
> > https://github.com/PaddlePaddle/PaddleOCR/blob/862b4d46edc55b8bf3d7806e7c316e11a1d5f478/configs/rec/PP-OCRv3/ch_PP-OCRv3_rec_distillation.yml#L182
> 
> 和这个配置文件一样，batch_size_per_card: 128，num_workers: 4

刚刚试了下，从128改为100确实能跑起来了，显存占用19349MiB /  24564MiB。这个配置文件是给多卡训练用的所以用的128吗？

**Reply by SWHL** [reply]
这个参数`batch_size_per_card`应该是单卡的batch_size。如果有多卡，默认其他卡也是这个数。
单卡和多卡配置文件都是这个

**Comment by SWHL** [comment]
好奇为啥不用v4的训练呢？一般旧版本，维护的就少了

**Reply by Alanhzl** [reply]
因为需要用到安卓设备上，我目前看到的文档好像是需要转为nb模型的，V4我尝试了下转不了，不清楚是不是paddlepaddle版本原因。

**Reply by SWHL** [reply]
建议先多尝试v4的，毕竟v3的不怎么维护了。
v4的遇到问题，大家也好一起商量解决

**Reply by Alanhzl** [reply]
好的，我之后尝试下V4,。在我的实际应用场景中，机器的配置会很低，且对识别时间有一定要求。V4的识别时间比V3,V2长很多的话，那我应该不能用V4了。

**Reply by SWHL** [reply]
嗯呢，我这里先行关闭Discussions了哈，有问题再重新开
